scripts/trim_indel_ends.py

```diff
@@ -0,0 +1,452 @@
+"""
+    File name: trim_indel_ends.py
+    Author: Yutong Qiu
+    Date: Nov 20
+    Description: For each pair of alinment between reads, if the best alignment includes indels at the beginning and ends
+        of the reads, then trim off the indels.
+    Input: 
+        - old sets of reads for each sample
+        - aln_stats
+    Output: 
+        - new set of reads for each sample.
+
+    Trimming logic alternatives:
+        - Trim prefix and suffix regardless of the rest of the alignment --- set a threshold
+        - Trim prefix and suffix only when the rest of the alignment is all matches --- set a threshold
+"""
+
+import pandas as pd
+from collections import defaultdict, Counter
+import re
+import edlib
+from Bio import SeqIO
+from Bio.SeqRecord import SeqRecord
+from Bio.Seq import Seq
+import matplotlib.pyplot as plt
+from tqdm import tqdm
+import os
+import sys
+import argparse
+import datetime
+
```

Comment:
# setup the logger

logging.getLogger(__name__).addHandler(logging.NullHandler())